src/platform_impl/windows/event_loop.rs

```diff
@@ -813,23 +813,28 @@ pub(super) unsafe extern "system" fn public_window_callback<T: 'static>(
             let initdata = createstruct.lpCreateParams as LONG_PTR;
             let initdata = &mut *(initdata as *mut InitData<'_, T>);
 
-            let runner = initdata.event_loop.runner_shared.clone();
-            if let Some((win, userdata)) = runner.catch_unwind(|| (initdata.post_init)(window)) {
-                initdata.window = Some(win);
-                let userdata = Box::into_raw(Box::new(userdata));
-                winuser::SetWindowLongPtrW(
-                    window,
-                    winuser::GWL_USERDATA,
-                    userdata as WindowLongPtr,
-                );
-                userdata
-            } else {
-                return -1;
-            }
+            let result = match initdata.on_nccreate(window) {
+                Some(userdata) => {
+                    winuser::SetWindowLongPtrW(window, winuser::GWL_USERDATA, userdata as _);
+                    winuser::DefWindowProcW(window, msg, wparam, lparam)
+                }
+                None => -1, // failed to create the window
+            };
+
+            return result;
         }
+        // `userdata` is set in `WM_NCCREATE` appearing before `WM_CREATE`.
```

Comment:
```suggestion
```
This comment isn't quite as necessary now that the call to `SetWindowLongPtrW` is clearly visible.